Display a message in motd Insights has not yet been used

stefwalter opened the pull request:
Insights represents key functionality of the RHEL operating system. Customers are missing out on Value of Subscription when not enabled. Thus we display a message in motd when Insights is installed but has never been used.

We use a ```/etc/motd.d/``` drop in file to do this. Once insights-client has been
used (and a new egg has been retrieved) we drop this message.

The message can be edited or removed in the configuration file:

    /etc/insights-client/insights-client.motd

This motd message seems to represent part of the insights-client command-line
RPM packaged wrapper of InsightsClient and is less suited to the core functionality.

andreasn commented on etc/insights-client.motd:
```diff
@@ -0,0 +1,2 @@
+Missing out on Insights benefits. To register run: insights-client --register
```
Needs to communicate the system status more clearly, and also the benefits of Insights.
https://www.redhat.com/en/technologies/management/insights is probably the place that has that written up best.

What about something like:
```
This system is not registered to Red Hat Insights.
https://www.redhat.com/en/technologies/management/insights

To register this system, run: insights-client --register
```

andreasn replied:
Possibly https://cloud.redhat.com is a better (and for sure much more stable URL).
This is what we went for in https://github.com/candlepin/subscription-manager/pull/2089

stefwalter replied:
Thanks. Updated the message.

stefwalter commented:
I've pushed new changes:

- Updated the message as suggested by @andreasn
- Added documentation to ```man insights-client``` on how to edit/remove the message

gravitypriest commented:
Hey @stefwalter, real quick -- is motd.d available in 8.0 or is this strictly an 8.1 and up feature?

stefwalter commented:
I see ```/etc/motd.d``` in both both RHEL 7.7 and RHEL 8.0 ... not sure about 6.x, but my guess would be that it's not available there.

gravitypriest commented:
No worries, all I needed to know. We're just building RHEL 8 for now so this'll be fine. Thanks for the PR.

gravitypriest commented:
Upon install and registration, running `rpm -V insights-client` complains that the motd file is missing from `/etc/motd.d`. Is this something we're okay with?

stefwalter commented:
```/etc/motd/insights-client``` is marked as a config file. @martinpitt do you think it's okay if a config file is marked as missing during ```rpm -V```?

If we'd prefer, we could replace it with a link to /dev/null instead of removing it (once insights-client has been used) and have the same affect.

gravitypriest commented:
That'd be preferable, I think. We just had to deal with some rpm -V errors in a previous release and don't want to start having more of them. In the interest of time I currently moved the symlink to post-install and took it out of the RPM-managed files, but that's not the most ideal solution.

martinpitt commented:
> do you think it's okay if a config file is marked as missing during `rpm -V`?

I'm afraid I'm not enough of an rpm packaging expert to give a definitive answer here.

> If we'd prefer, we could replace it with a link to /dev/null instead 

I do know that rpm has an irritating (for me, as I grew up with dpkg) tendency to put back deleted configuration files. Possibly the `%config(noreplace)` avoids that, but indeed having an explicit /dev/null symlink is more explicit and easier to understand IMHO. Then it's a changed configuration file instead of a removed one.

stefwalter commented:
Thinking about this further, an approach related to ```%post``` like @gravitypriest has taken may also be necessary. This is because of upgrades. We don't want this motd file to just appear on upgrades.

stefwalter commented:
@gravitypriest What do you think about #88
